### ISSUE TITLE
E0562 (Abstract return types are not allowed in variable binding) is not detected 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9381]
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4888-222-nightly
* **Rust toolchain version:** 1.66.0-nightly (a37499ae6 2022-09-18) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2.3 (CL-222.4167.35)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
<img width="933" alt="image" src="https://user-images.githubusercontent.com/49211026/191251565-943c9420-c562-45a2-9b5d-6213f818d515.png">

In the meta issue this error is marked as supported.

## Steps to reproduce
```rust
use crate::inner::T;

mod inner {
    pub trait T {
        type Output;
    }
}

fn func(_: impl T<Output=u32>) {} // normal usage

fn main() {
    let t: impl T<Output = u8>; // error
}
```
